cluster_test.go

```diff
@@ -1671,6 +1671,37 @@ func (s *S) TestMaxSocketUseTimeExpireAfterRelease(c *C) {
 	c.Assert(stats.SocketsExpired, Equals, 1)
 }
 
+// test to ensure that expired sockets are removed from active sockets list
+func (s *S) TestSocketExpiryEnsureSocketClosed(c *C) {
```

Comment:
The test tries to mimic what happened in production. Here is what happened to prod-canary,
- We've poollimit of x connections with y amount of socket expiry time. 
- Sockets that were getting expired were not properly released and that left active sockets list keep growing.
- The moment we hit the poollimit, go routines started to halt and wait for sockets to become available

This took about ~6h to reach. Looking at the code and interfaces I dont see a nice way to mock or to test this scenario. The actual code fix is pretty easy but trying to reproduce the scenario with given interface is not clean.